src/js/DropContainer.jsx

```diff
@@ -0,0 +1,284 @@
+import React from 'react';
+import classNames from 'classnames';
+import accepts from 'attr-accept';
+import css from '../stylesheets/dropcontainer.css';
+import UploadButton from './UploadButton';
+import ImagePreview from './ImagePreview';
+
+export default class DropContainer extends React.Component {
+   /**
+      * Base Container
+      * @param  {Object} props
+   */
+   constructor (props) {
+      super(props);
+
+      this.state = {
+         fileSrc: '',
+         dropzoneStyle: '',
+         isDragActive: false,
+         visibility: 'show'
+      };
+      this.onDragOver = this.onDragOver.bind(this);
+      this.onDrop = this.onDrop.bind(this);
+      this.onDragLeave = this.onDragLeave.bind(this);
+      this.onDragEnter = this.onDragEnter.bind(this);
+      this.handleChange = this.handleChange.bind(this);
+      this.handleInputButtonClick = this.handleInputButtonClick.bind(this);
+   }
+
+   componentDidMount () {
+      this.enterCounter = 0;
+   }
+
+   /**
+      * Handle the drag enter event
+      * @param  {Event]} event
+   */
+   onDragEnter (event) {
+      event.preventDefault();
+
+      // Count the dropzone and any children that are entered.
+      ++this.enterCounter;
+
+      // This is tricky. During the drag even the dataTransfer.
+      // files is null But Chrome implements some drag store,
+      // which is accesible via dataTransfer.items
+      const dataTransferItems = this.extractData(event),
+
+      // Now we need to convert the DataTransferList to Array
+      fileFormat = this.fileFormatFinder(this.props.accept),
```

Comment:
You're missing `var`, `let` or `const` here which means this ends up in the global scope. This could leads to weird behavior if something else does the same thing. Let's switch it to local scope instead.